crates/extensions/c8y_mapper_ext/src/operations/firmware_update.rs

```diff
@@ -104,22 +103,41 @@ impl CumulocityConverter {
                     .with_retain()
                     .with_qos(QoS::AtLeastOnce);
 
-                let metadata_topic = self.mqtt_schema.topic_for(
+                let cmd_metadata_topic = self.mqtt_schema.topic_for(
                     topic_id,
                     &Channel::CommandMetadata {
                         operation: OperationType::FirmwareUpdate,
                     },
                 );
-                let metadata_payload = FirmwareMetadata {
+
+                let update_cmd_metadata = Message::new(&cmd_metadata_topic, "{}")
+                    .with_retain()
+                    .with_qos(QoS::AtLeastOnce);
+
+                let twin_metadata_topic = self.mqtt_schema.topic_for(
+                    topic_id,
+                    &Channel::EntityTwinData {
+                        fragment_key: "firmware".to_string(),
+                    },
+                );
+
+                let twin_metadata_payload = FirmwareInfo {
                     name: Some(response.name.clone()),
                     version: Some(response.version.clone()),
                     remote_url: Some(response.remote_url.clone()),
                 };
-                let update_metadata = Message::new(&metadata_topic, metadata_payload.to_json())
-                    .with_retain()
-                    .with_qos(QoS::AtLeastOnce);
 
-                vec![c8y_notification, clear_local_cmd, update_metadata]
+                let twin_metadata =
+                    Message::new(&twin_metadata_topic, twin_metadata_payload.to_json())
+                        .with_retain()
+                        .with_qos(QoS::AtLeastOnce);
+
+                vec![
+                    c8y_notification,
+                    clear_local_cmd,
+                    update_cmd_metadata,
+                    twin_metadata,
```

Comment:
These 3 message are enough

```suggestion
                    c8y_notification,
                    clear_local_cmd,
                    twin_metadata,
```